src/api/Controllers/MigrateController.cs

```diff
@@ -49,44 +52,35 @@ public async Task<IActionResult> RecalculateCoordinates(bool onlyMissing = true,
-        }
+        logger.LogInformation(
+            "Starting recalculating coordinates with the following options: onlyMissing=<{OnlyMissing}>; dryRun=<{DryRun}>", dryRun, onlyMissing);
 
-        logger.LogInformation("Started recalculation coordinates with the following options: onlyMissing=<{OnlyMissing}>; dryRun=<{DryRun}>", dryRun, onlyMissing);
         var transformedCoordinates = 0;
 
         try
         {
             foreach (var borehole in context.Boreholes)
             {
-                if (borehole.OriginalReferenceSystem == ReferenceSystem.LV95)
-                {
-                    if (onlyMissing && borehole.LocationXLV03 != null && borehole.LocationYLV03 != null) continue;
-
-                    var (easting, northing) = await TransformCoordinatesAsync(
-                        "lv95tolv03", borehole.LocationX, borehole.LocationY).ConfigureAwait(false);
-                    borehole.LocationXLV03 = easting;
-                    borehole.LocationYLV03 = northing;
-                }
-                else
-                {
-                    if (onlyMissing && borehole.LocationX != null && borehole.LocationY != null) continue;
-
-                    var (easting, northing) = await TransformCoordinatesAsync(
-                        "lv03tolv95", borehole.LocationXLV03, borehole.LocationYLV03).ConfigureAwait(false);
-                    borehole.LocationX = easting;
-                    borehole.LocationY = northing;
-                }
+                var transformDirection = borehole.OriginalReferenceSystem == ReferenceSystem.LV95 ? Lv95ToLv03 : Lv03ToLv95;
+                var sourceLocationX = transformDirection == Lv95ToLv03 ? borehole.LocationX : borehole.LocationXLV03;
+                var sourceLocationY = transformDirection == Lv95ToLv03 ? borehole.LocationY : borehole.LocationYLV03;
+                var destinationLocationX = transformDirection == Lv95ToLv03 ? borehole.LocationXLV03 : borehole.LocationX;
+                var destinationLocationY = transformDirection == Lv95ToLv03 ? borehole.LocationYLV03 : borehole.LocationY;
+                var setDestinationLocationX = (double x) => transformDirection == Lv95ToLv03 ? borehole.LocationXLV03 = x : borehole.LocationX = x;
+                var setDestinationLocationY = (double y) => transformDirection == Lv95ToLv03 ? borehole.LocationYLV03 = y : borehole.LocationY = y;
+
+                if (onlyMissing && destinationLocationX != null && destinationLocationY != null) continue;
+
+                var reframeOptions = FormattableString.Invariant(
+                    $"{transformDirection}?easting={sourceLocationX}&northing={sourceLocationY}&format=json");
+
+                using var response = await httpClient.GetAsync(new Uri(reframeOptions, UriKind.Relative)).ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+
+                var jsonResult = await response.Content.ReadFromJsonAsync<JsonElement>().ConfigureAwait(false);
```

Comment:
Q/PP: Kann man das vereinfachen, in dem man `GetFromJsonAsync` verwendet anstatt `GetAsync` und dann `ReadFromJsonAsync`?

Reply:
Ja das würde gehen. Allerdings ist dann `EnsureSuccessStatusCode` nicht verfügbar und müsste dann wieder separat gehandelt werden.